hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -1001,8 +1001,13 @@ protected RequestFactory createRequestFactory() {
     try {
       storageClass = StorageClass.fromValue(storageClassConf);
     } catch (IllegalArgumentException e) {
-      LOG.warn("Unknown storage class property {}: {}; falling back to default storage class",
-          STORAGE_CLASS, storageClassConf);
+      if (storageClassConf.length() == 0) {
```

Comment:
Can do ``storageClassConf.isEmpty()`` rather than calculating the length.
BTW. If it is empty you need not to attempt `` storageClass = StorageClass.fromValue(storageClassConf);`` and then land up with this Exception handling

Reply:
Sounds good, much cleaner approach